rustls/src/error/mod.rs

```diff
@@ -979,6 +979,8 @@ pub enum PeerMisbehaved {
     OfferedIncorrectCompressions,
     PskExtensionMustBeLast,
     PskExtensionWithMismatchedIdsAndBinders,
+    ReceivedIllegalAlertLevel(AlertDescription),
+    ReceivedIllegalWarningAlert(AlertDescription),
```

Comment:
Nit: in the interest of conciseness, maybe skip the `Received` which kinda intuitively makes sense?

Also on the `Illegal` token, could we make do with something that is more common in our error stack already, like `Invalid`, `Unsupported` or `Unknown`? Or do you view all of these as semantically distinct? (If so, perhaps we should write that up somewhere.)